interoperator/controllers/multiclusterdeploy/sfserviceinstancereplicator/sfserviceinstancereplicator_controller.go

```diff
@@ -250,14 +256,59 @@ func (r *InstanceReplicator) Reconcile(ctx context.Context, req ctrl.Request) (c
 			"replicaState", replicaState, "replicaLastOperation", replicaLastOperation)
 	}
 
-	if lastErr != nil {
+	if lastErr != nil || reconcileErr != nil{
 		// re que if service/plan replication failed
 		return ctrl.Result{}, err
 	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *InstanceReplicator) reconcileSFPlan(instance *osbv1alpha1.SFServiceInstance, clusterID string, deleted bool) error {
+	ctx := context.Background()
+	serviceID := instance.Spec.ServiceID
+	planID := instance.Spec.PlanID
+	log := r.Log.WithValues("clusterID", clusterID, "serviceID", serviceID, "planID", planID)
+
+	var lastErr error
+
+	plan := &osbv1alpha1.SFPlan{}
+	planKey := types.NamespacedName{
+		Name:      planID,
+		Namespace: constants.InteroperatorNamespace,
+	}
+
+	err := r.Get(ctx, planKey, plan)
+	log.Info("Checking deletion eligibility for the plan", "is deleted: ", deleted, "is deletion ts for the plan set", !plan.GetDeletionTimestamp().IsZero())
```

Comment:
```suggestion
	log.Info("Checking deletion eligibility for the plan", "is instance deleted: ", deleted, "is deletion timestamp for the plan set: ", !plan.GetDeletionTimestamp().IsZero())
```
I feel it will be good to expand the acronyms. I might get confused checking the logs later 😄

Reply:
Done.